### ISSUE TITLE
maint-3.8: build failure: error: ‘_1’ was not declared in this scope

### ISSUE THREAD

**Issue by aleksander0m** [sandialabs/gr-pdu_utils#9]
I'm building gr-pdu_utils 3.8 in an Arch Linux environment:

```
aleksander@ares:~$ g++ --version
g++ (GCC) 11.1.0
Copyright (C) 2021 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

aleksander@ares:~$ pacman -Q | grep boost
boost 1.76.0-1
boost-libs 1.76.0-1
```

The build failed like this:
```
[  2%] Building CXX object lib/CMakeFiles/gnuradio-pdu_utils.dir/pdu_gmsk_fc_impl.cc.o
/home/aleksander/Development/gr-pdu_utils/lib/pdu_gmsk_fc_impl.cc: In constructor ‘gr::pdu_utils::pdu_gmsk_fc_impl::pdu_gmsk_fc_impl(float, std::vector<float>)’:
/home/aleksander/Development/gr-pdu_utils/lib/pdu_gmsk_fc_impl.cc:53:70: error: ‘_1’ was not declared in this scope
   53 |                     boost::bind(&pdu_gmsk_fc_impl::handle_pdu, this, _1));
      |                                                                      ^~
/home/aleksander/Development/gr-pdu_utils/lib/pdu_gmsk_fc_impl.cc:53:70: note: suggested alternatives:
In file included from /usr/include/boost/mpl/aux_/include_preprocessed.hpp:37,
                 from /usr/include/boost/mpl/placeholders.hpp:43,
                 from /usr/include/boost/iterator/iterator_categories.hpp:16,
                 from /usr/include/boost/iterator/iterator_adaptor.hpp:14,
                 from /usr/include/boost/iterator/reverse_iterator.hpp:10,
                 from /usr/include/boost/range/reverse_iterator.hpp:21,
                 from /usr/include/boost/range/rend.hpp:19,
                 from /usr/include/boost/foreach.hpp:82,
                 from /usr/include/gnuradio/basic_block.h:33,
                 from /usr/include/gnuradio/block.h:27,
                 from /home/aleksander/Development/gr-pdu_utils/lib/../include/pdu_utils/pdu_gmsk_fc.h:13,
                 from /home/aleksander/Development/gr-pdu_utils/lib/pdu_gmsk_fc_impl.h:15,
                 from /home/aleksander/Development/gr-pdu_utils/lib/pdu_gmsk_fc_impl.cc:14:
/usr/include/boost/mpl/aux_/preprocessed/gcc/placeholders.hpp:29:16: note:   ‘mpl_::_1’
   29 | typedef arg<1> _1;
      |                ^~
In file included from /usr/include/boost/smart_ptr/detail/shared_count.hpp:37,
                 from /usr/include/boost/smart_ptr/shared_ptr.hpp:17,
                 from /usr/include/boost/shared_ptr.hpp:17,
                 from /usr/include/pmt/pmt.h:30,
                 from /home/aleksander/Development/gr-pdu_utils/lib/../include/pdu_utils/constants.h:15,
                 from /home/aleksander/Development/gr-pdu_utils/lib/pdu_gmsk_fc_impl.h:14,
                 from /home/aleksander/Development/gr-pdu_utils/lib/pdu_gmsk_fc_impl.cc:14:
/usr/include/c++/11.1.0/functional:225:34: note:   ‘std::placeholders::_1’
  225 |     extern const _Placeholder<1> _1;
      |                                  ^~
In file included from /usr/include/boost/mpl/aux_/include_preprocessed.hpp:37,
                 from /usr/include/boost/mpl/placeholders.hpp:43,
                 from /usr/include/boost/iterator/iterator_categories.hpp:16,
                 from /usr/include/boost/iterator/iterator_adaptor.hpp:14,
                 from /usr/include/boost/iterator/reverse_iterator.hpp:10,
                 from /usr/include/boost/range/reverse_iterator.hpp:21,
                 from /usr/include/boost/range/rend.hpp:19,
                 from /usr/include/boost/foreach.hpp:82,
                 from /usr/include/gnuradio/basic_block.h:33,
                 from /usr/include/gnuradio/block.h:27,
                 from /home/aleksander/Development/gr-pdu_utils/lib/../include/pdu_utils/pdu_gmsk_fc.h:13,
                 from /home/aleksander/Development/gr-pdu_utils/lib/pdu_gmsk_fc_impl.h:15,
                 from /home/aleksander/Development/gr-pdu_utils/lib/pdu_gmsk_fc_impl.cc:14:
/usr/include/boost/mpl/aux_/preprocessed/gcc/placeholders.hpp:29:16: note:   ‘mpl_::_1’
   29 | typedef arg<1> _1;
      |                ^~
In file included from /usr/include/boost/bind/bind.hpp:2336,
                 from /usr/include/boost/thread/detail/thread.hpp:33,
                 from /usr/include/boost/thread/thread_only.hpp:22,
                 from /usr/include/boost/thread/thread.hpp:12,
                 from /usr/include/gnuradio/thread/thread.h:31,
                 from /usr/include/gnuradio/basic_block.h:31,
                 from /usr/include/gnuradio/block.h:27,
                 from /home/aleksander/Development/gr-pdu_utils/lib/../include/pdu_utils/pdu_gmsk_fc.h:13,
                 from /home/aleksander/Development/gr-pdu_utils/lib/pdu_gmsk_fc_impl.h:15,
                 from /home/aleksander/Development/gr-pdu_utils/lib/pdu_gmsk_fc_impl.cc:14:
/usr/include/boost/bind/placeholders.hpp:58:38: note:   ‘boost::placeholders::_1’
   58 | BOOST_STATIC_CONSTEXPR boost::arg<1> _1;
      |                                      ^~
In file included from /usr/include/boost/mpl/aux_/include_preprocessed.hpp:37,
                 from /usr/include/boost/mpl/placeholders.hpp:43,
                 from /usr/include/boost/iterator/iterator_categories.hpp:16,
                 from /usr/include/boost/iterator/iterator_adaptor.hpp:14,
                 from /usr/include/boost/iterator/reverse_iterator.hpp:10,
                 from /usr/include/boost/range/reverse_iterator.hpp:21,
                 from /usr/include/boost/range/rend.hpp:19,
                 from /usr/include/boost/foreach.hpp:82,
                 from /usr/include/gnuradio/basic_block.h:33,
                 from /usr/include/gnuradio/block.h:27,
                 from /home/aleksander/Development/gr-pdu_utils/lib/../include/pdu_utils/pdu_gmsk_fc.h:13,
                 from /home/aleksander/Development/gr-pdu_utils/lib/pdu_gmsk_fc_impl.h:15,
                 from /home/aleksander/Development/gr-pdu_utils/lib/pdu_gmsk_fc_impl.cc:14:
/usr/include/boost/mpl/aux_/preprocessed/gcc/placeholders.hpp:29:16: note:   ‘mpl_::_1’
   29 | typedef arg<1> _1;
      |                ^~
make[2]: *** [lib/CMakeFiles/gnuradio-pdu_utils.dir/build.make:76: lib/CMakeFiles/gnuradio-pdu_utils.dir/pdu_gmsk_fc_impl.cc.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:280: lib/CMakeFiles/gnuradio-pdu_utils.dir/all] Error 2
make: *** [Makefile:146: all] Error 2
```

I have patched this locally using `std` placeholders, see https://github.com/aleksander0m/gr-pdu_utils/commit/462cb3724ea52e112471d54376aa75106aa539e2, but not even sure that's the correct way to do it, even if it works.

Setting `-DBOOST_BIND_GLOBAL_PLACEHOLDERS` and `using namespace boost::placeholders;` may be a better way to handle this.

The build may also fail in the same way for the 3.9 branch or for master, didn't really investigate much more.

**Comment by dlitvinenko** [comment]
same in 3.10

**Comment by samwhiting-snl** [comment]
Addressed in #17, thanks.
